routing/pathfind.go

```diff
@@ -298,34 +293,25 @@ func newRoute(amtToSend, feeLimit lnwire.MilliSatoshi, sourceVertex Vertex,
 		// If this is the last hop, then we send the exact amount and
 		// pay no fee, as we're paying directly to the receiver, and
 		// there're no additional hops.
-		amtToForward := runningAmt
+		amtToForward := amtToSend
 		fee := lnwire.MilliSatoshi(0)
 
 		// If this isn't the last hop, to add enough funds to pay for
 		// transit over the next link.
 		if i != len(pathEdges)-1 {
 			// We'll grab the edge policy and per-hop payload of
 			// the prior hop so we can calculate fees properly.
-			prevEdge := pathEdges[i+1]
-			prevHop := route.Hops[i+1]
-
-			// The fee for this hop, will be based on how much the
-			// prior hop carried, as we'll need to increase the
-			// amount of satoshis incoming into this hop to
-			// properly pay the required fees.
-			prevAmount := prevHop.AmtToForward
-			fee = computeFee(prevAmount, prevEdge.ChannelEdgePolicy)
-
-			// With the fee computed, we increment the total amount
-			// as we need to pay this fee. This value represents
-			// the amount of funds that will come _into_ this edge.
-			runningAmt += fee
-
-			// Otherwise, for a node to forward an HTLC, then
-			// following inequality most hold true:
-			//
-			//     * amt_in - fee >= amt_to_forward
-			amtToForward = runningAmt - fee
+			nextHop := route.Hops[i+1]
```

Comment:
This is actually the _previous_ hop. We're traversing backwards right now along the route.

Reply:
It gets confusing with different definitions of prev and next. Is it next loop-wise or next in the payment route. It would be good to adhere to one definition. Discussed with @halseth and opted for "next = next in route".

Reply:
Maybe I should rename it to that. nextHopInRoute.

Reply:
Ah, yeah I was thinking "next = next in route". It is definitely confusing suing `prev` when we are raversing backwards. Looking at the comments, `lastHop` is also being used, so I think sticking to "first = first in route, last = last in route, next = next in route" makes sense. Comments will need to be updated to reflect this though.